src/platform/plugins/private/presentation_panel/public/panel_component/panel_header/use_hover_actions_styles.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { useEuiTheme } from '@elastic/eui';
+import { css } from '@emotion/react';
+
+import { useMemo } from 'react';
+
+export const useHoverActionStyles = (isEditMode: boolean, showBorder?: boolean) => {
+  const { euiTheme } = useEuiTheme();
+
+  const containerStyles = useMemo(() => {
+    const editModeOutline = `${euiTheme.border.width.thin} dashed ${euiTheme.colors.borderBaseFormsControl}`;
+    const viewModeOutline = `${euiTheme.border.width.thin} solid ${euiTheme.colors.borderBasePlain}`;
```

Comment:
We have plans to move this outside of the `PresentationPanel` entirely, where the component becomes agnostic to view mode + other Dashboard-specific things. However, that is out of the scope here - this cleanup does make this future work easier, though.